src/DocumentDB.ChangeFeedProcessor.IntegrationTests/SmokeTest.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Documents.ChangeFeedProcessor.FeedProcessing;
+using Microsoft.Azure.Documents.ChangeFeedProcessor.IntegrationTests.Utils;
+using Microsoft.Azure.Documents.ChangeFeedProcessor.PartitionManagement;
+using Microsoft.Azure.Documents.Client;
+using Xunit;
+
+namespace Microsoft.Azure.Documents.ChangeFeedProcessor.IntegrationTests
+{
+    public class SmokeTest
+    {
+        [Fact]
+        public async Task WriteToDocDB_SouldGeneratesIncomingEvent_IfStartingFromDefaultPoint()
+        {
+            string sessionId = Guid.NewGuid().ToString();
+
+            IntegrationTestsHelper.GetConfigurationSettings(out DocumentCollectionInfo feedCollectionInfo, out DocumentCollectionInfo leaseCollectionInfo, out int feedOfferThroughput, out int leaseOfferThroughput);
+            feedCollectionInfo.CollectionName = Guid.NewGuid().ToString();
+            leaseCollectionInfo.CollectionName = feedCollectionInfo.CollectionName + "-lease";
+
+            var feedClient = new DocumentClient(feedCollectionInfo.Uri, feedCollectionInfo.MasterKey, feedCollectionInfo.ConnectionPolicy);
+            await IntegrationTestsHelper.CreateDocumentCollectionAsync(feedClient, feedCollectionInfo.DatabaseName, new DocumentCollection { Id = feedCollectionInfo.CollectionName }, 400);
+
+            using (var client = new DocumentClient(leaseCollectionInfo.Uri, leaseCollectionInfo.MasterKey, leaseCollectionInfo.ConnectionPolicy))
+            {
+                await IntegrationTestsHelper.CreateDocumentCollectionAsync(client, leaseCollectionInfo.DatabaseName, new DocumentCollection { Id = leaseCollectionInfo.CollectionName }, 400);
+            }
+
+            TaskCompletionSource<bool> documentReceived = new TaskCompletionSource<bool>();
+            Task Process(IChangeFeedObserverContext changeFeedObserverContext, IReadOnlyList<Document> readOnlyList)
+            {
+                if (readOnlyList.Any(d => d.GetPropertyValue<string>("body") == sessionId))
+                {
+                    documentReceived.SetResult(true);
+                }
+                return Task.CompletedTask;
+            }
+
+            TaskCompletionSource<bool> partitionOpened = new TaskCompletionSource<bool>();
+            FeedProcessing.IChangeFeedObserverFactory observerFactory = new DelegatingMemoryObserverFactory(async ctx =>
+            {
+                await Task.Yield();
```

Comment:
it was a timing issue but I made it more explicit. I removed this Yield (it's strange here).

the issue was that I generated document before starting the query so the query didnt catch it. I did it explicitely like this because I didnt want to start reading from beginning, just to have such test because all our tests are reading from beginning. wdyt?